doxygen/contributor_help_pages/require_meta.md

```diff
@@ -49,135 +285,141 @@ class a_class<T, require_std_vector_st<is_var, T>> {
+    Ex:
 
-The container `requires` template parameters have an ending at their signature of _vt and _st to symbolize whether you want to inspect the @ref stan::value_type or @ref stan::scalar_type.
-A function that accepts eigen matrices with floating point value types can be defined as
+    ```
+    // Always decay types coming into the requires
+    template <typename T>
+    require_autodiff_t = require_t<is_autodiff<std::decay_t<T>>>;
+    ```
 
-```cpp
- template <typename Mat1, typename Mat2,
-   require_all_eigen_vt<std::is_floating_point, Mat1, Mat2>* = nullptr>
- auto add(Mat1&& A, Mat2&& B) { return A + B;}
-```
+* `_vt` uses `Check` to test the type `T` passed in and uses
+  `InnerCheck` to test the @ref value_type of `T`
 
-A function that accepts standard vectors of Eigen vectors whose scalar type is @ref stan::math::var types can be defined as
+    ```
+    template <template <class...> class TypeCheck, class... Check>   
+    require_std_vector_vt = require_vt<is_std_vector, TypeCheck, std::decay_t<Check>...>;
 
-```cpp
- template <typename Vec1, typename Vec2,
-   require_all_std_vector_vt<is_eigen_vector, Vec1, Vec2>* = nullptr,
-   require_all_std_vector_st<is_var, Vec1, Vec2>* = nullptr>
- auto add(Vec1&& A, Vec2&& B) {
-   std::vector<decltype<A[0] + B[0]>> return_vec;
-   std::transform(A.begin(), A.end(), B.begin(), return_vec.begin(),
-     [](auto&& x, auto&& y) {
-         return x + y;
-     });
-   return return_vec;
- }
-```
+    // Ex: Used to define a signature for `std::vectors` with a value type that is autodiffable
+    template <typename StdVec, require_std_vector_vt<is_var, StdVec>* = nullptr>
+    auto my_func(StdVec&& vec);
+    ```
 
-There are also `requires` template parameters for generically checking if a type's @ref stan::value_type or @ref stan::scalar_type is correct.
-To differentiate them from the Eigen and standard library vector checks the `vt` and `st` come *before* the type such as `require_vt_var<T>` which checks if a type `T`'s @ref stan::value_type satisfies @ref stan::is_var.
+* `_st` uses `Check` to test the type `T` passed in and uses
+  `InnerCheck` to test the @ref scalar_type of `T`
 
-The `requires` template parameters type traits allow Stan to have more generic types so that the library can forward Eigen expression and have better move semantics.
-For instance, the code below will accept any arbitrary Eigen expression that, if it's an rvalue, can be forwarded to another function.
+    ```
+    template <template <class...> class TypeCheck, class... Check>   
+    require_std_vector_st = require_st<is_std_vector, TypeCheck, std::decay_t<Check>...>;
+
+    // Ex: Used to define a signature for `std::vectors` with a scalar type that is autodiffable
+    template <typename StdVec, require_std_vector_st<is_var, StdVec>* = nullptr>
+    auto my_func(StdVec&& vec);
+    ```
```

Comment:
```suggestion
```cpp
template <template <class...> class TypeCheck, class... Check>   
require_std_vector_st = require_st<is_std_vector, TypeCheck, std::decay_t<Check>...>;

// Ex: Used to define a signature for `std::vectors` with a scalar type that is autodiffable
template <typename StdVec, require_std_vector_st<is_var, StdVec>* = nullptr>
auto my_func(StdVec&& vec);
```
```